app/models/user.rb

```diff
@@ -123,6 +123,16 @@ def censoring?(tag)
     self.censored_tags.exists?(name: tag)
   end
 
+  def censoring_tags?(tags)
+    tags.map! do |tag|
+      if tag.instance_of?(Tag)
+        tag = tag.name
+      end
+    end
+
+    self.censored_tags.pluck(:name) & tags
```

Comment:
```suggestion
    self.censored_tags.pluck(:name) & (tag_names)
```
上のように非破壊メソッドで置き換えた場合です。あと、ここでなんで&が出てきてるかよく分かってないので、もしよかったら意図を教えてください！あとで手元でも動かして試してみます。

Reply:
`self.censored_tags.pluck(:name)`がユーザーの検閲タグで、引数で渡ってくる`tags`がヌイートに紐付いたタグなので、和集合を取って検閲されているタグのみを取得する処理です
改めて見るとどんな処理なのか分かりづらいので`self.censored_tags.pluck(:name).select do |tag| tags.include?(tag) end`と書き直しても良いかも知れないです